cEP-0002.md

```diff
@@ -69,41 +69,40 @@ backwards compatibility. There will be new versions of `LocalBear` and
 `GlobalBear` that will be called `FileWiseBear` and `ProjectWideBear`
 respectively.
 
-# The File Proxy
+# The File Factory
 
-With the new bear design we are also moving to a new wrapper object to wrap
-every file. The file proxy is an object that contains properties about the
-file currently processed (such as the filename). It contains:
+With the new bear design we are also moving to a new factory pattern
+for loading files.
 
-+ The filename, this is the starting point for every other attribute.
-+ The content of the file as a string.
-+ The content of the file, line-splitted.
+The factory instance is located at `coalib.input.file.factory`,
+and will default to an instance of a class `LineCacheFileFactory` 
+that uses `linecache` to provide access to file contents.
 
-Note: Other properties can be included later as and when required.
+Additional FileFactory implementations will be created as required
+to suit various workloads.
 
-A prototype of the file proxy:
-
-```python
-from coala_utils.decorators import generate_eq
+Alternative factories may be used by altering the module attribute
+`factory`.  Switching factories should be done before any files have been
+loaded by the default factory, however any existing file objects already
+loaded by the default factory must remain usable for the duration of
+the file object lifetime.
 
+A factory access function `coalib.input.file.open` will be
+completely polymorphic with the regular `open()` builtin function,
+excepting that the returned content handler will implement a new
+abstract class `FileReader`, which only implements methods `readline`
+and `readlines`, and has an attribute `name` containing the absolute
```

Comment:
readlines and readline would only be for the LineCacheFileFactory described above - correct ?

Also, rather than having readline() and readlines(), maybe it'd be simpler to have `readlines(num_lines=None)` where if `num_lines` is 0, it gives the whole file.